tests/e2e-instrumentation/instrumentation-python-musl/00-install-collector.yaml

```diff
@@ -0,0 +1,22 @@
+apiVersion: opentelemetry.io/v1alpha1
```

Comment:
Adding specific e2e tests is not needed because the `e2e-test-app-python` docker image is already based on alpine right? looks like tests are failing atm

Reply:
>  looks like tests are failing on main

Can you elaborate this a bit more?


> Adding specific e2e tests is not needed because the e2e-test-app-python docker image is already based on alpine right? 

Maybe this was something where I failed. Our idea is, at some point, add verifications to know if the libraries were injected properly and verify they are emitting data.

Reply:
BTW the other thing that should be kept in sync is the Python version of the two instrumentation because the ABI changes between python versions.

Reply:
> At the moment the `e2e-test-app-python` is based on alpine but the python instrumentation image is glibc based. This is a problem because binary extensions are not portable between different C libraries (among other incompatibilities). So this PR builds them and copies one for musl or glibc depending on the configuration.

Didn't notice this when I added the images. As mentioned in the previous comment, the idea is to add real E2E checking if the instrumentation is generating real data. Since we are not checking this, issues like the one you saw are happening. This is something we need to fix with that image.

You can reuse that image (since it is `musl` based) for your E2E test. We need to add a new one for `glibc`.

Reply:
> You can reuse that image (since it is `musl` based) for your E2E test. We need to add a new one for `glibc`.

I'm already using that image in the musl e2e :+1: